Source/Libraries/openXDA.Model/Channels/ChannelData.cs

```diff
@@ -289,6 +309,67 @@ private static byte[] ToData(List<DataPoint> data, int seriesID)
             return returnArray;
         }
 
+        private static byte[] ToDigitalData(List<DataPoint> data, int seriesID)
+        {
+            int i;
+            IEnumerable<DataPoint> changeSeries = data.Where((point,index) => index == 0 || index == data.Count - 1 || point.Value != data[index-1].Value);
+
+            const ushort NaNValue = ushort.MaxValue;
+            const ushort MaxCompressedValue = ushort.MaxValue - 1;
+            double range = changeSeries.Select(item => item.Value).Max() - changeSeries.Select(item => item.Value).Min();
+            double decompressionOffset = changeSeries.Select(item => item.Value).Min();
+            double decompressionScale = range / MaxCompressedValue;
+            double compressionScale = (decompressionScale != 0.0D) ? 1.0D / decompressionScale : 0.0D;
+
+            long startTime = data.First().Time.Ticks;
+            long sampleRate = data[1].Time.Ticks -  startTime;
+
+            int totalByteLength = 2 * sizeof(int) + 2 * sizeof(long) + 2 * sizeof(double) + changeSeries.Count() * (sizeof(int) + sizeof(ushort));
+
+            byte[] result = new byte[totalByteLength];
+            int offset = 0;
+
+            offset += LittleEndian.CopyBytes(data.Count, result, offset);
+            offset += LittleEndian.CopyBytes(seriesID, result, offset);
+
+            offset += LittleEndian.CopyBytes(startTime, result, offset);
+            offset += LittleEndian.CopyBytes(sampleRate, result, offset);
+
+            offset += LittleEndian.CopyBytes(decompressionOffset, result, offset);
+            offset += LittleEndian.CopyBytes(decompressionScale, result, offset);
+
+            foreach (DataPoint dataPoint in changeSeries)
+            {
+                ushort compressedValue = (ushort)Math.Round((dataPoint.Value - decompressionOffset) * compressionScale);
+
+                if (compressedValue == NaNValue)
+                    compressedValue--;
+
+                if (double.IsNaN(dataPoint.Value))
+                    compressedValue = NaNValue;
+
+                int time = (int)(dataPoint.Time.Ticks - startTime);
```

Comment:
The full range of possible values here is less than 5 minutes.
`2^31-1 ticks = 2147483647 ticks / 10000000 ticks per second ~= 215 seconds = 3.5 minutes`

Perhaps this doesn't matter because PQubes don't produce digitals, but if we ever encounter a COMTRADE file that lasts more than 3.5 minutes, this format will wrap around to negative timestamps and cause some problems for visualization.